core/src/main/java/org/apache/calcite/avatica/AvaticaStatement.java

```diff
@@ -268,18 +278,20 @@ protected void close_() {
   }
 
   public int getMaxFieldSize() throws SQLException {
-    throw connection.HELPER.unsupported();
+    checkOpen();
+    return 0;
   }
 
   public void setMaxFieldSize(int max) throws SQLException {
-    throw connection.HELPER.unsupported();
+    throw AvaticaConnection.HELPER.unsupported();
```

Comment:
Would it make sense to make this be a noop instead of returning unsupported now that `getMaxFieldSize()` is set to always return 0.

Reply:
not if the value is set to something different from 0, if not supported

Reply:
Ok that makes sense. Thanks!